test/ipns-pubsub.js

```diff
@@ -0,0 +1,172 @@
+/* eslint-env mocha */
+'use strict'
+
+const chai = require('chai')
+const dirtyChai = require('dirty-chai')
+const expect = chai.expect
+chai.use(dirtyChai)
+
+const { fromB58String } = require('multihashes')
+const base64url = require('base64url')
+const ipns = require('ipns')
+
+const parallel = require('async/parallel')
+const retry = require('async/retry')
+const series = require('async/series')
+
+const DaemonFactory = require('ipfsd-ctl')
+
+const waitFor = require('./utils/wait-for')
+
+const config = {
+  Addresses: {
+    API: '/ip4/0.0.0.0/tcp/0',
+    Gateway: '/ip4/0.0.0.0/tcp/0',
+    Swarm: []
+  }
+}
+
+const namespace = '/record/'
+
+const spawnJsDaemon = (callback) => {
+  DaemonFactory.create({ type: 'js' })
+    .spawn({
+      disposable: true,
+      initOptions: { bits: 512 },
+      args: ['--enable-namesys-pubsub'], // enable ipns over pubsub
+      config
+    }, callback)
+}
+
+const spawnGoDaemon = (callback) => {
+  DaemonFactory.create()
+    .spawn({
+      disposable: true,
+      initOptions: { bits: 1024 },
+      args: ['--enable-namesys-pubsub'], // enable ipns over pubsub
+      config
+    }, callback)
+}
+
+const ipfsRef = '/ipfs/QmPFVLPmp9zv5Z5KUqLhe2EivAGccQW2r7M7jhVJGLZoZU'
+
+describe('ipns-pubsub', () => {
+  let nodeAId
+  let nodeBId
+  let nodes = []
+
+  // Spawn daemons
+  before(function (done) {
+    // CI takes longer to instantiate the daemon, so we need to increase the timeout
+    this.timeout(80 * 1000)
+
+    series([
+      (cb) => spawnGoDaemon(cb),
+      (cb) => spawnJsDaemon(cb)
+    ], (err, daemons) => {
+      expect(err).to.not.exist()
+      nodes = daemons
+
+      done()
+    })
+  })
+
+  // Get node ids
+  before(function (done) {
+    this.timeout(100 * 1000)
+
+    parallel([
+      (cb) => nodes[0].api.id(cb),
+      (cb) => nodes[1].api.id(cb)
+    ], (err, ids) => {
+      expect(err).to.not.exist()
+      expect(ids).to.exist()
+      expect(ids[0].id).to.exist()
+      expect(ids[1].id).to.exist()
+
+      nodeAId = ids[0]
+      nodeBId = ids[1]
+
+      nodes[0].api.swarm.connect(ids[1].addresses[0], () => {
+        setTimeout(done, 60000) // wait for republish as we can receive the republish message first
```

Comment:
This is such a long wait, it would be great if this was somehow configurable...

@Stebalien is it possible in go-ipfs?

@vasco-santos perhaps add a `console.log` line here to inform the user this pause is happening...?

Reply:
`go-ipfs` already accepts the `republish` configuration through the node config. For JS Land, it is one of the things that I want to integrate during this quarter.

Meanwhile, I can add a `console.log` here!

Reply:
Oh wait, `go-ipfs` allows the `RepublishPeriod` and `RecordLifetime`, but not the time for the first `InitialRebroadcastDelay`, afaik!